app/uk/gov/hmrc/hec/services/FileCreationService.scala

```diff
@@ -147,12 +148,14 @@ class FileCreationServiceImpl @Inject() (timeProvider: TimeProvider) extends Fil
     case None                     => SAStatusMapping(None, None)
   }
 
-  private def ctStatusMapping(ctStatus: Option[CTStatus]) = ctStatus match {
-    case Some(CTStatus.ReturnFound)        => CTStatusMapping(Some('Y'), None, Some('Y'))
-    case Some(CTStatus.NoticeToFileIssued) => CTStatusMapping(Some('N'), Some('Y'), Some('Y'))
-    case Some(CTStatus.NoReturnFound)      => CTStatusMapping(Some('N'), Some('N'), Some('Y'))
-    case None                              => CTStatusMapping(None, None, Some('N'))
-  }
+  private def ctStatusMapping(ctStatus: Option[CTStatus]) =
+    ctStatus match {
+      case Some(CTStatus.ReturnFound)        => CTStatusMapping(Some('Y'), None, Some('Y'))
+      case Some(CTStatus.NoticeToFileIssued) => CTStatusMapping(Some('N'), Some('Y'), Some('Y'))
+      case Some(CTStatus.NoReturnFound)      => CTStatusMapping(Some('N'), Some('N'), Some('Y'))
+      case None                              => CTStatusMapping(None, None, Some('Y'))
```

Comment:
it'd be a bit safer to make the answer here when it's `None` also look at whether or not an accounting period end date exists but maybe passing in `Option[LocalDate]` or something to this function. In theory, you can have no accounting period and so there'd still no CT status in which case you'd want `N` here

Reply:
I have taken care of that scenario at liken number [175](https://github.com/hmrc/hec/blob/ba8fcedce0df2fddd4cf4db04ec2419b2509eb8a/app/uk/gov/hmrc/hec/services/FileCreationService.scala#L175)

Reply:
There is a test case for this scenario as well [here](https://github.com/hmrc/hec/blob/ba8fcedce0df2fddd4cf4db04ec2419b2509eb8a/test/uk/gov/hmrc/hec/services/FileCreationServiceSpec.scala#L598) where there i no accounting period and in file hasAccounting period is N

Reply:
ah ok - I guess looking at that function in isolation is where it looks dangerous. Not sure if there's a way to make it more obvious, even if it's just maybe calling the method something like `ctStatusMappingAccountingPeriodExists`

Reply:
there's also another new requirement we need to include here - on the digital journey, even if we know the actual CT status, if they've said they weren't chargeable then we should come into this `CTStatusMapping(None, None, Some('Y'))` case and omit the return info being supplied in the above cases. This means we may need to pass in the value of `notChargeble` to this function somehow

Reply:
hmm, i can do that change in here.

Reply:
[updated](https://github.com/hmrc/hec/pull/56/commits/552c10c65cec6068aca89350c73b8d08d7f0c86e)